src/mesh/interpolation/bilinear.cxx

```diff
@@ -28,7 +28,11 @@
 #include <vector>
 
 Bilinear::Bilinear(int y_offset, Mesh *mesh)
-  : Interpolation(y_offset, mesh), w0(localmesh), w1(localmesh), w2(localmesh), w3(localmesh) {
+  : Interpolation(y_offset, mesh),
+  w0(localmesh, CELL_CENTRE, DIRECTION::X, DIRECTION::YOrthogonal, DIRECTION::Z),
+  w1(localmesh, CELL_CENTRE, DIRECTION::X, DIRECTION::YOrthogonal, DIRECTION::Z),
+  w2(localmesh, CELL_CENTRE, DIRECTION::X, DIRECTION::YOrthogonal, DIRECTION::Z),
+  w3(localmesh, CELL_CENTRE, DIRECTION::X, DIRECTION::YOrthogonal, DIRECTION::Z) {
```

Comment:
Do these need explicit values?

Reply:
In the present implementation, we have to set explicit values here, because these fields need to be allocated before the `ParallelTransform` has finished being constructed, so we can't get the default `yDirectionType` from the `ParallelTransform` without creating a circular call. Because this class is only used by `FCITransform` we know what the default should be, so we can set it explicitly.